spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/catalog/GravitinoCatalog.java

```diff
@@ -336,4 +351,36 @@ private String getDatabase(NameIdentifier gravitinoIdentifier) {
         "Only support 3 level namespace," + gravitinoIdentifier.namespace());
     return gravitinoIdentifier.namespace().level(2);
   }
+
+  @VisibleForTesting
+  static com.datastrato.gravitino.rel.TableChange transformTableChange(TableChange change) {
+    if (change instanceof TableChange.AddColumn) {
+      TableChange.AddColumn addColumn = (TableChange.AddColumn) change;
+      return com.datastrato.gravitino.rel.TableChange.addColumn(
+          addColumn.fieldNames(),
+          SparkTypeConverter.toGravitinoType(addColumn.dataType()),
+          addColumn.comment(),
+          transformColumnPosition(addColumn.position()),
+          addColumn.isNullable());
+    } else if (change instanceof TableChange.DeleteColumn) {
+      TableChange.DeleteColumn deleteColumn = (TableChange.DeleteColumn) change;
+      return com.datastrato.gravitino.rel.TableChange.deleteColumn(
+          deleteColumn.fieldNames(), deleteColumn.ifExists());
+    } else {
+      throw new UnsupportedOperationException(
+          String.format("Unsupported table change %s", change.getClass().getName()));
+    }
+  }
+
+  private static com.datastrato.gravitino.rel.TableChange.ColumnPosition transformColumnPosition(
+      TableChange.ColumnPosition columnPosition) {
```

Comment:
we should return `defaultPos` only when columnPosition is null, if its' not First or After, let's throw a exception. in case some SparkVersion supports Before, it will be error if return `defaultPos`.